csharp/src/Google.Protobuf/Collections/RepeatedField.cs

```diff
@@ -124,11 +124,49 @@ public void AddEntriesFrom(CodedInputStream input, FieldCodec<T> codec)
             }
         }
 
+#if GOOGLE_PROTOBUF_SUPPORT_SPAN
+        /// <summary>
+        /// Adds the entries from the given input stream, decoding them with the specified codec.
+        /// </summary>
+        /// <param name="input">The input stream to read from.</param>
+        /// <param name="codec">The codec to use in order to read each entry.</param>
+        public void AddEntriesFrom(ref CodedInputReader input, FieldCodec<T> codec)
+        {
+            // TODO: Inline some of the Add code, so we can avoid checking the size on every

```

Comment:
I'm still thinking of some good way to avoid the duplication. Having basically two copies of the same code is not good for maintainability. And the chances are both copies of the code will be optimized over time (e.g. see the TODO). 
Ideally I'd like to avoid situation where all changes to protobuf codebase need to be basically done twice, because that sounds like a recipe to end up with bugs in the future.